src/components/TaskCard.tsx

```diff
@@ -0,0 +1,74 @@
+import { Pencil, Trash2 } from "lucide-react";
+import { useState } from "react";
+import { createPortal } from "react-dom";
+import { useTasksStore } from "../stores/tasks";
+import { Task } from "../types/Task";
+import TaskForm from "./TaskForm";
+
+export default function TaskCard({ task }: { task: Task }) {
+  const [showModal, setShowModal] = useState(false);
+  const deleteTask = useTasksStore((state: any) => state.deleteTask);
+
+  const handleDelete = (taskId: string) => {
+    const existingTasks = JSON.parse(localStorage.getItem("tasks") || "[]");
```

Comment:
You are using this multiple times, do you think it's better define this in your tasks store?

const persistedTasksInStorage = JSON.parse(localStorage.getItem("tasks") || "[]");

Reply:
I know where you are going, you are right, I'll check it out